src/lettuce/plugins/welcome.py

```diff
@@ -0,0 +1,29 @@
+from machine.plugins.base import MachineBasePlugin
+from machine.plugins.decorators import process
+
+
+class WelcomePlugin(MachineBasePlugin):
+    @process("member_joined_channel")
```

Comment:
As per Donnie's suggestion, this should be listening to [team_join](https://api.slack.com/events/team_join) event.